keyboard-navigation.js.tid

```diff
@@ -192,61 +213,56 @@ document.onkeyup = function(e) {
 		}
 	}
 	var last_tiddler_index = tiddler_index;
-	if (j_released || k_released) {
+
+	if (down_key_released || up_key_released || other_bind_used) {
 		if (tiddler_index < 0 || (!isInViewport(tiddlers[tiddler_index]) && !isElementCloseToTop(tiddlers[tiddler_index]))) {
 			tiddler_index = findTopmostTiddler(tiddlers);
-			if (j_released) tiddler_index -= 1;
-		}
-		if (j_released) {
-			// go down
-			tiddler_index += 1;
-			if (tiddler_index >= tiddlers.length) tiddler_index = cycle ? 0 : tiddlers.length - 1;
-		} else  {  // e.code == "KeyK"
-			// go up
-			tiddler_index -= 1;
-			if (tiddler_index < 0) tiddler_index = cycle ? tiddlers.length - 1 : 0;
-		}
-		if (Date.now() - timestamp_last_navigation < 800) { // milliseconds
-			// if user navigates quickly, jump instantly to next or previous tiddler
-			tiddlers[tiddler_index].scrollIntoView();
-		} else {
-			// scroll smoothly to next or previous tiddler
-			tiddlers[tiddler_index].scrollIntoView({behavior: "smooth", block: "start", inline: "nearest"});
-		}
-		timestamp_last_navigation = Date.now();
-		if (MARK_CURRENT_TIDDLER) {
-			if (last_tiddler_index >= 0)
-				tiddlers[last_tiddler_index].classList.remove("activeTiddler");
-			tiddlers[tiddler_index].classList.add("activeTiddler");
+			if (down_key_released) tiddler_index -= 1;
 		}
-		getActiveTiddlersTitle();
-	} else if (c_released) {
-		// close current tiddler
-		if (tiddlers[tiddler_index] === undefined) return;
-		if (!isInViewport(tiddlers[tiddler_index])) return;
-		var button = tiddlers[tiddler_index].getElementsByClassName("tc-btn-%24%3A%2Fcore%2Fui%2FButtons%2Fclose")[0];
-		if (!button) return;
-		button.click();
-		if (tiddlers.length == 1) return;  // no tiddler left (after closing the only one)
-		// the next tiddler becomes the active one; if the last one was closed, the one before will be activated
-		if (tiddler_index >= tiddlers.length - 1) {
-			tiddler_index = tiddlers.length - 2;
-			if (MARK_CURRENT_TIDDLER)
+		if (down_key_released || up_key_released) {
+			if (down_key_released) {
+				// go down
+				tiddler_index += 1;
+				if (tiddler_index >= tiddlers.length) tiddler_index = cycle ? 0 : tiddlers.length - 1;
+			} else if (up_key_released) {
+				// go up
+				tiddler_index -= 1;
+				if (tiddler_index < 0) tiddler_index = cycle ? tiddlers.length - 1 : 0;
+			}
+			if (Date.now() - timestamp_last_navigation < INSTANT_NAVIGATION_BEFORE || INSTANT_NAVIGATION) {
```

Comment:
Just for efficiency. If you want instant navigation, there is no need to check the current time on every `j`/`k` keystroke:

```suggestion
			if (INSTANT_NAVIGATION || Date.now() - timestamp_last_navigation < INSTANT_NAVIGATION_BEFORE ) {
```